dataland-frontend/tests/component/components/pages/LandingPage.cy.ts

```diff
@@ -0,0 +1,126 @@
+import { checkButton, checkImage, checkLinkByContent, checkLinkByTarget } from "@ct/testUtils/ExistenceChecks";
+import NewLandingPage from "@/components/pages/NewLandingPage.vue";
+import { minimalKeycloakMock } from "@ct/testUtils/Keycloak";
+
+describe("Component test for the landing page", () => {
+  it("Check if essential elements are present", () => {
+    cy.mountWithPlugins(NewLandingPage, {
+      keycloak: minimalKeycloakMock({
+        authenticated: false,
+      }),
+    }).then(() => {
+      validateTopBar();
+      validateIntroSection();
+      validateBrandsSection();
+      // TODO test I AM INTERESTED button
+      // TODO test get in contact button
+
+      validateQuotesSlides();
+      validateHowItWorksSlides();
+      checkNewFooter();
+    });
+  });
+});
+
+/**
+ * Validates the elements of the top bar
+ */
+function validateTopBar(): void {
+  /**
+   * Gets the top bar element
+   * @returns the top bar element
+   */
+  function getTopBar(): Cypress.Chainable {
+    return cy.get("header");
+  }
+  checkImage("Dataland banner logo", "gfx_logo_dataland_orange_S.svg", getTopBar());
+  checkButton("signup_dataland_button", "Sign Up", getTopBar());
+  checkLinkByContent("Login", getTopBar());
+}
+
+/**
+ * Validates the elements of the intro section
+ */
+function validateIntroSection(): void {
+  checkImage("Liberate Data -  Empower Autonomy. Dataland, the Open ESG Data Platform.", "gfx_logo_d_orange_S.svg");
```

Comment:
You could import the text ("Liberate Data ...") from the content.json.

That way the test won't break if someone changes the content.json in the future.

What do you think?

Reply:
Lets discuss this.